dns/checkip/src/opnsense/mvc/app/controllers/OPNsense/DynDNS/Api/CheckipsettingsController.php

```diff
@@ -0,0 +1,264 @@
+ *    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ *    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+namespace OPNsense\DynDNS\Api;
+
+use OPNsense\Base\ApiMutableModelControllerBase;
+use OPNsense\Core\Config;
+use OPNsense\DynDNS\CheckIP_Service;
+use OPNsense\DynDNS\CheckIP_Factory_Default_Service;
+use OPNsense\Base\UIModelGrid;
+
+/**
+ * Class SettingsController
+ * @package OPNsense\CheckIP
+ */
+class CheckIPSettingsController extends ApiMutableModelControllerBase
+{
+    static protected $internalModelName = 'CheckIP';
+    static protected $internalModelClass = '\OPNsense\DynDNS\CheckIP_Service';
+
+    /**
+     * list with valid model node types
+     */
+    private $nodeTypes = array('service');
+
+    /**
+     * query CheckIP settings
+     * @param $nodeType
+     * @param $uuid
+     * @return result array
+     */
+    public function getAction($nodeType = null, $uuid = null)
+    {
+        $result = array("result" => "failed");
+        if ($this->request->isGet() && $nodeType != null) {
+            $this->validateNodeType($nodeType);
+            $mdlCheckIPService = new CheckIP_Service();
+            if ($uuid != null) {
+                $node = $mdlCheckIPService->getNodeByReference($nodeType . '.' . $uuid);
+            } else {
+                $node = $mdlCheckIPService->$nodeType->Add();
+            }
+            if ($node != null) {
+                $result['checkip'] = array($nodeType => $node->getNodes());
+                $result['result'] = 'ok';
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * set CheckIP properties
+     * @param $nodeType
+     * @param $uuid
+     * @return status array
+     */
+    public function setAction($nodeType = null, $uuid = null)
+    {
+        $result = array("result" => "failed", "validations" => array());
+        if ($this->request->isPost() && $this->request->hasPost("checkip") && $nodeType != null) {
+            $this->validateNodeType($nodeType);
+            $mdlCheckIPService = new CheckIP_Service();
+            if ($uuid != null) {
+                $node = $mdlCheckIPService->getNodeByReference($nodeType . '.' . $uuid);
+            } else {
+                $node = $mdlCheckIPService->$nodeType->Add();
+            }
+            if ($node != null) {
+                $checkIPInfo = $this->request->getPost("checkip");
+
+                $node->setNodes($checkIPInfo[$nodeType]);
```

Comment:
this looks dangerous due to missing validations

Reply:
There is validation there.  Is it not correct?
Added validation comment.